platform/engineering/design-docs/2020-05-26-content-validation.md

```diff
@@ -0,0 +1,278 @@
+
+Additionally, we have the preview server which shows content editors any accessibility violations, but they must deliberately go to the server from a link in Drupal to see this.
+
+#### Pain points
+
+##### Content Writers
+
+Content writers want to quickly draft, publish, and deploy content. However, `vets-website` is only deployed to production once per day. To allow content writers to move faster, there is a [partial deploy](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/getting-started/workflow/deploy/#partial-deploy--static-page-changes-only), which only includes static page changes (`vagov-content` and Drupal).
+
+No accessibility checking is performed on the content-only deploys. That means that content writers don't have a proactive way to learn if their content includes accessibility errors. Adding accessibility checking to the CMS is on the roadmap, but it doesn't currently exist.
+
+##### VA.gov Users
+
+To enable a faster publishing workflow, all tests (except link checking) are intentionally skipped during content-only deploys. That creates an opportunity for end users to encounter accessibility errors. Additionally, inconsistencies between how the CMS handles redirects and how `vets-website` handles redirects, it's possible for users to encounter broken links in production.
+
+##### FE Tools team
+
+Invalid content causes warning logs during staging deploys, and breaks the build during production deploys. These broken builds then need manual intervention, and require coordinating with the CMS team.
+
+##### Anybody that is trying to ship code for the VA.gov front end
+
+If invalid content is breaking the production build, and therefore blocking the production deploy, then no team on any contract is able to ship their frontend code to VA.gov.
+
+## High Level Design
+
+As part of the [build separation wor](https://github.com/department-of-veterans-affairs/va.gov-team/issues/2719)k, two separate pipelines will be created for for built code to be deployed to S3 to prevent failures in one from blocking between CMS and application deployments.
+
+The Metalsmith static content build will be extracted to the `content-build` repo and deleted from `vets-website`, and its deployment pipeline. Webpack will handle the application build for the `vets-website` repo, and Metalsmith will handle the static content build for the `content-build` repo.
+
+The validation of static content will take place in a scheduled job that runs every workday instead of ???. Since the broken link check happens during the Metalsmith build, and the accessibility check relies on the output of the content build, the scheduled job will run a script to produce a build that only has the steps necessary for content validation. That build will only be used for reporting broken links and accessibility errors, and will have no impact on deploys. Those reports will be delivered to the #cms-team channel via existing Slack integrations.
```

Comment:
Because the content-only deploy, AKA [partial deploy](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/getting-started/workflow/deploy#partial-deploy--static-page-changes-only), only runs the drupal broken link checks, and none of the `vets-website` broken link checks, we've already indirectly decided that fast content deploys are worth the risk of broken links landing in production. 

In the **Alternatives** section, I've included a [**Highest quality, but lowest speed** option](https://github.com/department-of-veterans-affairs/va.gov-team/blob/cfae598310780e75243d3b5f6cbad8f1031079aa/platform/engineering/design-docs/2020-05-26-content-validation.md#highest-quality-but-lowest-speed).